cmake/Platform/Core/Targets/ArduinoBootloaderBurnTargetCreator.cmake

```diff
@@ -28,93 +28,55 @@ function(create_arduino_bootloader_burn_target TARGET_NAME BOARD_ID PROGRAMMER P
-        message("Missing bootloader image, not creating bootloader burn target ${BOOTLOADER_TARGET}.")
+    if (NOT EXISTS "${ARDUINO_BOOTLOADERS_PATH}/${BOOTLOADER_PATH}/${BOOTLOADER_FILE_NAME}")
+        message("Missing bootloader image '${ARDUINO_BOOTLOADERS_PATH}/${BOOTLOADER_PATH}/${BOOTLOADER_FILE_NAME}', not creating bootloader burn target ${BOOTLOADER_TARGET}.")
         return()
     endif ()
 
+    #check for required bootloader parameters
+    foreach (ITEM lock_bits unlock_bits high_fuses low_fuses)
+        #do not make fatal error if field doesn't exists, just don't create bootloader burn target
+        _get_board_property_if_exists(${BOARD_ID} bootloader.${ITEM} BOOTLOADER_${ITEM})
+        if (NOT BOOTLOADER_${ITEM})
+            message("Missing bootloader.${ITEM}, not creating bootloader burn target ${BOOTLOADER_TARGET}.")
+            return()
+        endif ()
+    endforeach ()
+
     # Erase the chip
     list(APPEND AVRDUDE_ARGS "-e")
 
     # Set unlock bits and fuses (because chip is going to be erased)
-
-    if (${BOARD_ID}.bootloader.unlock_bits)
-        list(APPEND AVRDUDE_ARGS "-Ulock:w:${${BOARD_ID}.bootloader.unlock_bits}:m")
-    else ()
-        # Arduino 1.6 SDK
-        list(APPEND AVRDUDE_ARGS
-                "-Ulock:w:${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.unlock_bits}:m")
-    endif ()
-
-    if (${BOARD_ID}.bootloader.extended_fuses)
-        list(APPEND AVRDUDE_ARGS "-Uefuse:w:${${BOARD_ID}.bootloader.extended_fuses}:m")
-    elseif (${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.extended_fuses})
-        list(APPEND AVRDUDE_ARGS
-                "-Uefuse:w:${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.extended_fuses}:m")
-    endif ()
-    if (${BOARD_ID}.bootloader.high_fuses)
-        list(APPEND AVRDUDE_ARGS
-                "-Uhfuse:w:${${BOARD_ID}.bootloader.high_fuses}:m"
-                "-Ulfuse:w:${${BOARD_ID}.bootloader.low_fuses}:m")
-    else ()
-        list(APPEND AVRDUDE_ARGS
-                "-Uhfuse:w:${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.high_fuses}:m"
-                "-Ulfuse:w:${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.low_fuses}:m")
-    endif ()
-
-    # Set bootloader image
-    list(APPEND AVRDUDE_ARGS "-Uflash:w:${${BOARD_ID}.bootloader.file}:i")
-
-    # Set lockbits
-    if (${BOARD_ID}.bootloader.lock_bits)
-        list(APPEND AVRDUDE_ARGS "-Ulock:w:${${BOARD_ID}.bootloader.lock_bits}:m")
-    else ()
-        list(APPEND AVRDUDE_ARGS
-                "-Ulock:w:${${BOARD_ID}.menu.cpu.${BOARD_MCU}.bootloader.lock_bits}:m")
-    endif ()
-
+    list(APPEND AVRDUDE_ARGS "-Ulock:w:${BOOTLOADER_unlock_bits}:m")
+    # extended fuses is optional
+    _get_board_property_if_exists(${BOARD_ID} bootloader.extended_fuses BOOTLOADER_extended_fuses)
+    if (BOOTLOADER_extended_fuses)
+        list(APPEND AVRDUDE_ARGS "-Uefuse:w:${BOOTLOADER_extended_fuses}:m")
+    endif()
+
+    list(APPEND AVRDUDE_ARGS
```

Comment:
Do you assume that all those property exist since you've iterated over them a few lines above?
If so than it's completely OK.

Reply:
Yes I only get here if all properties was correctly read in the loop.

Reply:
👍